Sources/SwiftDocC/Model/ParametersAndReturnValidator.swift

```diff
@@ -0,0 +1,473 @@
+        )
+    }
+    
+    /// Creates a validated parameter section containing only the parameters that exist in each language representation's function signature.
+    ///
+    /// - Parameters:
+    ///   - parameters: The symbol's documented parameters.
+    ///   - signatures: The symbol's containing only the values that exist in that language representation's function signature.
+    ///   - symbolKind: The documented symbol's kind, for use in diagnostics.
+    ///   - hasDocumentedReturnValues: `true` if the symbol has documented its return values; otherwise `false`.
+    /// - Returns: A parameter section variant containing only the parameters that exist in each language representation's function signature.
+    private func makeParametersSectionVariants(
+        _ parameters: [Parameter]?,
+        _ signatures: Signatures,
+        _ symbolKind: SymbolGraph.Symbol.Kind?,
+        hasDocumentedReturnValues: Bool
+    ) -> DocumentationDataVariants<ParametersSection> {
+        guard let parameters = parameters, !parameters.isEmpty else {
+            guard hasDocumentedReturnValues, Self.shouldAddObjectiveCErrorParameter(signatures, parameters) else {
+                // The symbol has no parameter documentation or return value documentation and none should be synthesized.
+                return DocumentationDataVariants(defaultVariantValue: nil)
+            }
+            // Since the symbol documented its return value synthesize an "error" parameter for it.
+            var variants = DocumentationDataVariants<ParametersSection>()
+            for trait in signatures.keys {
+                variants[trait] = ParametersSection(parameters: []) // Add an empty section so that this language doesn't fallback to the Objective-C content.
+            }
+            variants[.objectiveC] = ParametersSection(parameters: [Parameter(name: "error", contents: Self.objcErrorDescription)]) // This parameter is synthesized and doesn't have a source range.
+            return variants
+        }
+        
+        var variants = DocumentationDataVariants<ParametersSection>()
+        var allKnownParameterNames: Set<String> = []
+        var argumentLabelsAndPossibleParameterNames: [String: Set<String>] = [:]
+        // Group the parameters by name to be able to diagnose parameters documented more than once.
+        let parametersByName = [String: [Parameter]](grouping: parameters, by: \.name)
+        
+        for (trait, signature) in signatures {
+            // Collect all language representations's parameter information from the function signatures.
+            let knownParameterNames = Set(signature.parameters.map(\.name))
+            allKnownParameterNames.formUnion(knownParameterNames)
+            for parameter in signature.parameters {
+                if let externalName = parameter.externalName {
```

Comment:
Is `externalName` the second label that Swift functions can use, example match vs. foundDocumentationExtension in:

```
private func nodeWithInitializedContent(reference: ResolvedTopicReference, match foundDocumentationExtension: DocumentationContext.SemanticResult<Article>?) -> DocumentationNode {
```

Reply:
No, external name is the name that's included in the symbol name. I [documented this in the SymbolKit PR.](https://github.com/apple/swift-docc-symbolkit/pull/64/files#diff-38139c158e2d8026197306fe59fe789160efbcc944f94f773b2e674372c53527R36-R53)

> In Swift which supports separate argument labels and parameter names, this function has one parameter where ``externalName`` is "with" and ``name`` is "someValue".
>
>```swift
>func doSomething(with someValue: Int) {}
>//               │    ╰─ name
>//               ╰─ externalName
>```

Reply:
Yes - that's what I meant. Ok perfect thanks for the explanation.

Reply:
I updated the documentation in https://github.com/apple/swift-docc/pull/776/commits/ef264671245be85daf65bb1ab7a3eb96a6d8a95b to clarify the difference between parameter names and external parameter names.

Reply:
thank you looks great!